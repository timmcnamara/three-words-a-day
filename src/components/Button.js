import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { dark } from "../styles/styleGuide";

// Change the name of the styled component?
const Btn = styled.button`
  background: white;
  color: ${dark};
  font-size: 1em;
  margin: 1em;
  padding: 0.75em 2em;
  border: 2px solid ${dark};
  border-radius: 3px;
  text-transform: uppercase;
`;

// Or the name of the component?
const Button = props => <Btn>{props.title}</Btn>;

Button.propTypes = {
  title: PropTypes.string
};

export default Button;
