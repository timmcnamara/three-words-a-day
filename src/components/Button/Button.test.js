import { render, cleanup } from "react-testing-library";
import React from "react";
import Button from "./";
import "jest-dom/extend-expect";

const props = {
  text: "hello",
  onClick: jest.fn()
};

afterEach(cleanup);

describe("<Button />", () => {
  it("render button component", () => {
    const { asFragment } = render(<Button {...props} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("test a random function", () => {});

  it("renders text on button", () => {
    const { getByTestId } = render(<Button {...props} />);
    expect(getByTestId("button-global")).toHaveTextContent("hello");
  });

  it("calls onClick when user clicks", () => {
    const { getByTestId } = render(<Button {...props} />);

    getByTestId("button-global").click();

    expect(props.onClick).toHaveBeenCalledTimes(1);
  });
});
