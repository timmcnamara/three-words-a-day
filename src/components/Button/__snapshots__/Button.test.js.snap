// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> render button component 1`] = `
<DocumentFragment>
  <button
    class="sc-bdVaJa ljraLK"
    data-testid="button-global"
  >
    hello
  </button>
</DocumentFragment>
`;
